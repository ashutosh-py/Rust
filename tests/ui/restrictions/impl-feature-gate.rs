// gate-test-impl_restriction
// revisions: with_gate without_gate
//[with_gate] check-pass

#![crate_type = "lib"]
#![cfg_attr(with_gate, feature(impl_restriction))]

pub impl(crate) trait Bar {} //[without_gate]~ ERROR

mod foo {
    pub impl(in foo) trait Baz {} //[without_gate]~ ERROR
}
