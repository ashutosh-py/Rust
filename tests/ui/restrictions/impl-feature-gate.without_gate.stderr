error[E0658]: restrictions are experimental
  --> $DIR/impl-feature-gate.rs:8:5
   |
LL | pub impl trait Foo {}
   |     ^^^^
   |
   = help: add `#![feature(impl_restriction)]` to the crate attributes to enable

error[E0658]: restrictions are experimental
  --> $DIR/impl-feature-gate.rs:9:5
   |
LL | pub impl(crate) trait Bar {}
   |     ^^^^^^^^^^^
   |
   = help: add `#![feature(impl_restriction)]` to the crate attributes to enable

error[E0658]: restrictions are experimental
  --> $DIR/impl-feature-gate.rs:12:9
   |
LL |     pub impl(in foo) trait Baz {}
   |         ^^^^^^^^^^^^
   |
   = help: add `#![feature(impl_restriction)]` to the crate attributes to enable

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0658`.
